Server/htdocs/AppController/commands_RSM/utilities/RSMbadgesManagement.php

```diff
@@ -25,69 +35,51 @@ function RSupdateBadgeForUser($userID, $clientID) {
     return $results;
 }
 
+
 // This function generates a random string of the given length
 function generateRandomString($length = 10) {
-    // This is the list of characters allowed inside the generated string
-    $characters = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    $allowedCharacters = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
     $randomString = '';
 
     for ($i = 0; $i < $length; $i++) {
-        // We loop until the desired length is reached, adding random characters
-        $randomString .= $characters[random_int(0, strlen($characters) - 1)];
+        $randomString .= $allowedCharacters[random_int(0, strlen($allowedCharacters) - 1)];
     }
 
-    // And return the result
     return $randomString;
 }
 
 
-function RSgetUniqueBadge(){
-    /* ***************************************************************************************
-    DESCRIPTION
-    Create a new badge that does not exist in the list of user badges in the database.
-
-    PARAMETERS
-    It does not need any parameters.
-
-    RETURN
-    badge: The badge itself, as a 32-character string (MD5 hash)
-    ***************************************************************************************
-    */
-
-    // We will use this variable in order to control if the new badge exists or not
+// We can restrict the action to a single clientID
+function RSgetUniqueBadge($RSclientID = null){
     $exists = false;
 
     do {
-        // Let's generate a badge
         $badge = md5(generateRandomString(256));
-
-        // Ask the database for badges like the new one
-        $results = RScountBadge($badge);
-
-        // Obtain the data from the query
-        if ($results) $result = $results->fetch_assoc();
         
-        // Check if we found a badge like ours in the database
-        if ($result['total'] <> 0) $exists = true; // The badge is already stored in the database. We must generate a new one
+        $results = RSbadgeExist($badge, $RSclientID);
+        if ($results == true) {
+            $exists = true;
+        }
```

Comment:
Creo que en este bucle sobran variables. Se puede usar la variable results en vez de exists

Reply:
One of the variables is left over and we simplify the loop. We have tested its behaviour in the development environment.

`// We can restrict the action to a single clientID
function RSgetUniqueBadge($RSclientID = null){
    do {
        $badge = md5(generateRandomString(256));
    } while (RSbadgeExist($badge, $RSclientID));

    // If the execution reaches this point, the badge does not exist so we can return it
    return $badge;
}`